### PR TITLE
Use xcm-handler to execute XCM locally

### LINKED ISSUE #399: Use xcm-handler to execute XCM locally
Now `xcm-handler` can execute XCM locally with an `origin` parameter (via [this PR](https://github.com/paritytech/cumulus/pull/357)), we don't have to use the `xcm-executor` workaround.

### PR DESCRIPTION
**Pull request by shaunxw**
Closes #399

### CODE REVIEW

File: xcm-support/src/lib.rs [hunk 1]
```diff
@@ -12,10 +12,14 @@ use sp_std::{
...
 };
 
-use xcm::v0::{Error, Junction, MultiAsset, MultiLocation, Result};
+use xcm::v0::{Error, Junction, MultiAsset, MultiLocation, Result, Xcm};
 use xcm_executor::traits::{FilterAssetLocation, LocationConversion, MatchesFungible, NativeAsset, TransactAsset};
 
-use frame_support::{log, traits::Get};
+use frame_support::{dispatch::DispatchResult, log, traits::Get};
+
+pub trait XcmHandler<AccountId> {
```
Comment (xlc) [inline xcm-support/src/lib.rs:20]: shouldn't this be in cumulus / polkadot?

File: xcm-support/src/lib.rs [hunk 2]
```diff
@@ -12,10 +12,14 @@ use sp_std::{
...
+	fn execute_xcm(origin: AccountId, xcm: Xcm) -> DispatchResult;
```
Comment (xlc) [inline xcm-support/src/lib.rs:21]:
```suggestion
	fn execute_xcm(origin: &AccountId, xcm: Xcm) -> DispatchResult;
```

File: xcm-support/src/lib.rs (hunk 1 above)
Comment (shaunxw) [reply xcm-support/src/lib.rs:20]: There is no usage of this trait in Cumulus, not sure if it's good to add it there. We can probably open a PR to add it later.

File: xcm-support/src/lib.rs (hunk 2 above)
Comment (shaunxw) [reply xcm-support/src/lib.rs:21]: The `xcm-handler` requires `AccountId` param.

Making it as reference would require `Clone` bound on `AccountId` generic and do clones.

**Review by xlc — APPROVED** [review]

**Pull request merged**